SIGSEGV: cli.setPolicyFromFlags(0x0, 0xc000369b20, 0x1013c53, 0x16)

So I've somehow got kopia into a state where running any `kopia policy set` command results in the following stack trace:

```
[root@ip-172-20-62-250 ~]#  kopia policy set --global --compression zstd
Setting policy for (global)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd4c076]

goroutine 1 [running]:
github.com/kopia/kopia/cli.setPolicyFromFlags(0x0, 0xc00032db20, 0x1013c53, 0x16)
        /root/kopia/cli/command_policy_set.go:109 +0x26
github.com/kopia/kopia/cli.setPolicy(0x11abe00, 0xc0000a7920, 0xc0002c8f00, 0x1, 0xc0002c8f00)
        /root/kopia/cli/command_policy_set.go:92 +0x272
github.com/kopia/kopia/cli.maybeRepositoryAction.func1.1(0x0, 0x0)
        /root/kopia/cli/app.go:113 +0x148
github.com/kopia/kopia/cli.withProfiling(...)
        /root/kopia/cli/profile_disabled.go:7
github.com/kopia/kopia/cli.maybeRepositoryAction.func1(0xc00015c090, 0x40be03, 0xeb83e0)
        /root/kopia/cli/app.go:92 +0x5d
gopkg.in/alecthomas/kingpin%2ev2.(*actionMixin).applyActions(0xc00034bd58, 0xc00015c090, 0x0, 0x0)
        /root/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/actions.go:28 +0x6d
gopkg.in/alecthomas/kingpin%2ev2.(*Application).applyActions(0xc0000e6690, 0xc00015c090, 0x0, 0x0)
        /root/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:557 +0xdc
gopkg.in/alecthomas/kingpin%2ev2.(*Application).execute(0xc0000e6690, 0xc00015c090, 0xc0000b92c0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x28f)
        /root/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:390 +0x8f
gopkg.in/alecthomas/kingpin%2ev2.(*Application).Parse(0xc0000e6690, 0xc0000b0130, 0x5, 0x5, 0x1, 0xc0000af478, 0x0, 0x1)
        /root/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:222 +0x1fe
main.main()
        /root/kopia/main.go:78 +0x178
```

I've tried `kopia cache clear` but it still occurs and it doesn't actually seem to matter what the arguments are (`kopia policy set --global --keep-hourly 0` crashes with the same stack trace).

`kopia policy get` works fine:

```
[root@ip-172-20-62-250 ~]#  kopia policy get --global
Policy for (global):

Retention:
  Annual snapshots:    0           (defined for this target)
  Monthly snapshots:  12           (defined for this target)
  Weekly snapshots:   12           (defined for this target)
  Daily snapshots:    14           (defined for this target)
  Hourly snapshots:    0           (defined for this target)
  Latest snapshots:   10           (defined for this target)

Files policy:
  Ignore rules:
    .snapshot/                     (defined for this target)
    LIBTEST/                       (defined for this target)
    cache.lck                      (defined for this target)
  Read ignore rules from files:
    .kopiaignore                   (defined for this target)

Error handling policy:
  Ignore file read errors:       false       (defined for this target)
  Ignore directory read errors:  false       (defined for this target)

Scheduled snapshots:
  None

Compression:
  Compressor: "zstd" (defined for this target)
  Compress files regardless of extensions.
  Compress files of all sizes.
```

A little more about the environment - I have 19 ec2 instances concurrently migrating backups from an older format to a kopia repository stored on s3. The kopia version is a master built from 2e7fe7c2b3f7f76de30130db3348d26eb68db809.

There is a `kopia policy set` command at the start of the script the instances run to ensure the repository has the correct settings, but possibly running this concurrently has broken things?

--- a/cli/command_policy_set.go
+++ b/cli/command_policy_set.go
@@ -82,6 +82,7 @@ func setPolicy(ctx context.Context, rep *repo.Repository) error {
 
 	for _, target := range targets {
 		p, err := policy.GetDefinedPolicy(ctx, rep, target)
+
 		switch {
 		case err == policy.ErrPolicyNotFound:
 			p = &policy.Policy{}

--- a/cli/command_policy_set.go
+++ b/cli/command_policy_set.go
@@ -82,8 +82,11 @@ func setPolicy(ctx context.Context, rep *repo.Repository) error {
 
 	for _, target := range targets {
 		p, err := policy.GetDefinedPolicy(ctx, rep, target)
-		if err == policy.ErrPolicyNotFound {
+		switch {
+		case err == policy.ErrPolicyNotFound:
 			p = &policy.Policy{}
+		case err != nil:
+			return errors.Wrap(err, "could not get defined policy")
 		}
 
 		printStderr("Setting policy for %v\n", target)

